test/README.md

```diff
@@ -1,13 +1,16 @@
-# Blah blah blah
+# Running Tests
 
-some words here...
-probably:
+To run the tests, perform the following steps:
 
-+ don't forget to substitute `<version>` in the [config.py](config.py)
-+ Ensure all OpenID options are valid
-+ Have `pytest` installed for OpenID tests
+1) Startup a fresh iRODS server.
+2) Startup the Testing keycloak image.
+3) Startup the iRODS HTTP API server.
+4) Ensure the HTTP API can communicate with the iRODS server and the keycloak image.
```

Comment:
Are these instructions meant to be context-free (that is, can be used in containers or not)?

My suggestion here would be to provide some example commands to make it easy to follow along.

Here's some stuff that I just made up, but sorta illustrates what I'm getting at:

1. Startup a fresh iRODS server (consider using the testing environment: \<link\>)
2. Startup the Testing keycloak image
```bash
docker run -d keycloak --name keycloak-container
```
3. Startup the iRODS HTTP API server (for instructions, see \<link\>)
4. Ensure the HTTP API can communicate with the iRODS server and the keycloak image
```bash
docker network create coolnetwork
docker network connect coolnetwork keycloak-container
docker network connect coolnetwork irods-container
docker network connect coolnetwork http-api-container
# something equivalent to "ils" to confirm that everything sees each other
```
Or something like that. Better still would be if we had a Compose project since nobody likes wiring together containers manually. This can of course happen in a later PR.

Reply:
> Are these instructions meant to be context-free (that is, can be used in containers or not)?

I try to be context-free, since I'm assuming that not everyone runs everything in containers.

> My suggestion here would be to provide some example commands to make it easy to follow along.

Maybe only show an example of containerized versions?

How about high level for those that don't use the containers, and step-by-step for those that do?

~What if this depends on the testing env? Do I just link that and say have at it?~
Just saw the example below.

> Here's some stuff that I just made up, but sorta illustrates what I'm getting at:
> 
>     1. Startup a fresh iRODS server (consider using the testing environment: <link>)
> 
>     2. Startup the Testing keycloak image
> 
> 
> ```shell
> docker run -d keycloak --name keycloak-container
> ```
> 
>     3. Startup the iRODS HTTP API server (for instructions, see <link>)
> 
>     4. Ensure the HTTP API can communicate with the iRODS server and the keycloak image
> 
> 
> ```shell
> docker network create coolnetwork
> docker network connect coolnetwork keycloak-container
> docker network connect coolnetwork irods-container
> docker network connect coolnetwork http-api-container
> # something equivalent to "ils" to confirm that everything sees each other
> ```
> 
> Or something like that. Better still would be if we had a Compose project since nobody likes wiring together containers manually. This can of course happen in a later PR.

Yes, compose would be nice. No idea if we can compose a compose, or if we need a flattened compose. Or in english, can we "import" a compose file, or does it all need to be in one?

There's also the "if we depend on the testing compose, how does that work?" question... Like do they need to be side by side directory wise? Should probably put this in a separate issue.

Reply:
> Or something like that. Better still would be if we had a Compose project since nobody likes wiring together containers manually. This can of course happen in a later PR.

Agreed. Let's handle docker compose later. The most important goal of this PR is to make building and running a keycloak docker image for the HTTP API easy.

> Maybe only show an example of containerized versions?

Yes. Docker is required.

> How about high level for those that don't use the containers, and step-by-step for those that do?

We don't have to cover different environments. Docker is required because that's how we test things.

> Yes, compose would be nice. No idea if we can compose a compose, or if we need a flattened compose. Or in english, can we "import" a compose file, or does it all need to be in one?
> 
> There's also the "if we depend on the testing compose, how does that work?" question... Like do they need to be side by side directory wise? Should probably put this in a separate issue.

We don't have to figure that out today. Those questions will be answered later, when we put energy into adding docker compose support.

Please open an issue to add support for docker compose.

Reply:
New write-up is looking good so far. Once we have an issue for the Docker Compose solution linked here, I think we can resolve this and hash out the new words in separate conversation(s)

Reply:
Issue created: #395

Resolving.